src/main/java/backend/resource/TurboLabel.java

```diff
@@ -91,26 +97,113 @@ private void ______METHODS______() {
         }
     }
 
-    private static String joinWith(String group, String name, boolean exclusive) {
-        return group + (exclusive ? EXCLUSIVE_DELIMITER : NONEXCLUSIVE_DELIMITER) + name;
+
+    /**
+     * Returns the first TurboLabel in labels that matches labelName
+     * Assumption: the labelName matches at least 1 TurboLabel
+     * @return TurboLabel that matches labelName
+     */
+    public static TurboLabel getFirstMatchingTurboLabel(List<TurboLabel> labels, String labelName) {
+        Optional<TurboLabel> firstMatchingLabel = getMatchedLabels(labels, labelName).stream().findFirst();
+        assert firstMatchingLabel.isPresent();
+        return firstMatchingLabel.get();
     }
 
-    public boolean isExclusive() {
-        return getDelimiter(actualName).isPresent() && getDelimiter(actualName).get().equals(EXCLUSIVE_DELIMITER);
+    public static List<String> getLabelsNameList(List<TurboLabel> labels) {
+        return labels.stream()
+                .map(TurboLabel::getFullName)
+                .collect(Collectors.toList());
     }
 
-    public boolean hasGroup() {
-        return getGroup().isPresent();
+    /**
+     * Returns the list of label names that matches a query
+     * Uses partial matching and matching is case-insensitive
+     * @param repoLabels
+     * @param nameQuery
+     * @return
+     */
+    public static List<TurboLabel> filterByNameQuery(List<TurboLabel> repoLabels, String nameQuery) {
+        return repoLabels
+                .stream()
+                .filter(label -> Utility.containsIgnoreCase(label.getShortName(), nameQuery))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Returns the list of label names that belongs to a group and matches a query
+     * Uses partial matching and matching is case-insensitive
+     * @param repoLabels
+     * @param groupQuery
+     * @return
+     */
+    public static List<TurboLabel> filterByGroupQuery(List<TurboLabel> repoLabels, String groupQuery) {
+        if (groupQuery.isEmpty()) return repoLabels;
+
+        return repoLabels
+                .stream()
+                .filter(label -> {
+                    if (label.isInGroup()) {
+                        return Utility.containsIgnoreCase(label.getGroupName(), groupQuery);
+                    }
+                    return false;
+                })
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Returns the list of labels that matches the keyword
+     * i.e. the label's group contains keyword's group and label's name contains keyword's name
+     * @param repoLabels
+     * @param keyword
+     * @return
+     */
+    public static List<TurboLabel> getMatchedLabels(List<TurboLabel> repoLabels, String keyword) {
+        TurboLabel query = new TurboLabel("", keyword);
```

Comment:
This is a misuse of a constructor. Instead, extract a utility method `splitLabelName(String labelName):String[]`?